examples/tv-app/android/java/OnOffManager.cpp

```diff
@@ -0,0 +1,112 @@
+/**
+ *
+ *    Copyright (c) 2022 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#include "OnOffManager.h"
+#include "TvApp-JNI.h"
+#include <app-common/zap-generated/attributes/Accessors.h>
+#include <app-common/zap-generated/ids/Clusters.h>
+#include <app/util/af.h>
+#include <jni.h>
+#include <lib/support/CHIPJNIError.h>
+#include <lib/support/JniReferences.h>
+#include <lib/support/JniTypeWrappers.h>
+
+using namespace chip;
+
+static constexpr size_t kOnffManagerTableSize = EMBER_AF_ON_OFF_CLUSTER_SERVER_ENDPOINT_COUNT;
+
+namespace {
+
+OnOffManager * gOnOffManagerTable[kOnffManagerTableSize] = { nullptr };
+
+}
+
+void emberAfOnOffClusterInitCallback(EndpointId endpoint)
+{
+    ChipLogProgress(Zcl, "TV Android App::OnOff::PostClusterInit");
+    TvAppJNIMgr().PostClusterInit(chip::app::Clusters::OnOff::Id, endpoint);
+}
+
+void OnOffManager::NewManager(jint endpoint, jobject manager)
+{
+    ChipLogProgress(Zcl, "TV Android App: OnOffManager::NewManager");
+    uint16_t ep = emberAfFindClusterServerEndpointIndex(static_cast<chip::EndpointId>(endpoint), app::Clusters::OnOff::Id);
+    VerifyOrReturn(ep != 0xFFFF && ep < EMBER_AF_ON_OFF_CLUSTER_SERVER_ENDPOINT_COUNT,
+                   ChipLogError(Zcl, "TV Android App::OnOff::NewManager: endpoint %d not found", endpoint));
+
+    OnOffManager * mgr = new OnOffManager();
```

Comment:
where is this memory freed? should we check if `gOnOffManagerTable[ep]` is already non-null and free it then or return an error?

This method does error checking on arguments, should it return success/failure?

Reply:
> where is this memory freed?
It will never be freed until app exit. Because onoff cluster will not be destoried

>should we check if gOnOffManagerTable[ep] is already non-null and free it then or return an error?
fixed

>This method does error checking on arguments, should it return success/failure?
added check and log